pkg/scuzz/sg_linux.go

```diff
@@ -112,24 +113,42 @@ type packet struct {
 	command commandDataBlock
 	status  statusBlock
 	block   dataBlock
+	word    wordBlock
 }
 
 // SGDisk implements a Disk using the Linux SG device
 type SGDisk struct {
 	f      *os.File
 	dev    uint8
 	packID uint32
-	master bool
+
+	// TimeOut is the timeout on a disk operation.
+	TimeOut time.Duration
 }
 
-// NewSGDisk returns a Disk that uses the Linux SCSI Generic Device
-func NewSGDisk(n string) (Disk, error) {
+// SGDiskOpt allows callers of NewSGDisk to set values
+type SGDiskOpt func(d *SGDisk)
+
+// NewSGDisk returns a Disk that uses the Linux SCSI Generic Device.
+// It also does an Identify to verify that the target name is a true
+// lba48 device.
+func NewSGDisk(n string, opt ...SGDiskOpt) (Disk, error) {
 	f, err := os.OpenFile(n, os.O_RDWR, 0)
 	if err != nil {
 		return nil, err
 	}
-	// TODO: figure out the dev and packID, or if we even need them.
-	return &SGDisk{f: f}, nil
+	s := &SGDisk{f: f}
+	if _, err := s.Identify(); err != nil {
+		return nil, err
+	}
+	s.TimeOut, err = time.ParseDuration("15000ms")
```

Comment:
`s.Timeout = 15 * time.Second`?

Reply:
Also, set the timeout before Identify?

Reply:
yeah. I blanked totally one time.Second for some reason. Fixed.